js/src/Ice/OutgoingConnectionFactory.js

```diff
@@ -85,15 +80,7 @@ export class OutgoingConnectionFactory {
             // received over such connections.
             //
             const adapter = routerInfo.getAdapter();
-            const defaultsAndOverrides = this._instance.defaultsAndOverrides();
             endpoints.forEach((endpoint) => {
```

Comment:
I believe there is a small bug here and in Java (and probably C#/C++):
https://github.com/zeroc-ice/ice/blob/61662730fbe8845c8c9641a828585c4caf484a88/java/src/Ice/src/main/java/com/zeroc/IceInternal/OutgoingConnectionFactory.java#L196

We need to clear the timeout of the endpoint (set it to -1) to search it correctly.

Reply:
BTW the compress also seems bogus since we put compress=true and compress=false in the connection list.

We should clean-up this compress handling (make it like endpoint timeout) in all languages in a follow-up PR.